packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -915,10 +918,16 @@ class HotRunner extends ResidentRunner {
     }
     await Future.wait(allDevices);
 
-    // Check if any isolates are paused.
+    globals.printTrace('Evicting dirty assets');
+    await _evictDirtyAssets();
+
+    // Check if any isolates are paused and reassemble those
+    // that aren't.
     final List<FlutterView> reassembleViews = <FlutterView>[];
+    final List<Future<void>> reassembleFutures = <Future<void>>[];
     String serviceEventKind;
     int pausedIsolatesFound = 0;
+    bool failedReassemble = false;
     for (final FlutterDevice device in flutterDevices) {
```

Comment:
I restructured this to make it easier to read and test - since I need a reference to device.vmService to use the new API